matrix-sdk/src/main/java/org/matrix/androidsdk/data/DataRetriever.java

```diff
@@ -159,82 +177,82 @@ public void run() {
+                                Event oldestEvent = store.getOldestEvent(roomId);
 
-                        if (events.chunk.size() != 0) {
-                            events.chunk.get(0).mToken = events.start;
+                                if (events.chunk.size() != 0) {
+                                    events.chunk.get(0).mToken = events.start;
 
-                            // there is no more data on server side
-                            if (null == events.end) {
-                                events.end = Event.PAGINATE_BACK_TOKEN_END;
-                            }
+                                    // there is no more data on server side
+                                    if (null == events.end) {
+                                        events.end = Event.PAGINATE_BACK_TOKEN_END;
+                                    }
+
+                                    events.chunk.get(events.chunk.size() - 1).mToken = events.end;
+
+                                    Event firstReturnedEvent = events.chunk.get(0);
+                                    if ((oldestEvent != null) && (firstReturnedEvent != null)
+                                            && TextUtils.equals(oldestEvent.eventId, firstReturnedEvent.eventId)) {
+                                        events.chunk.remove(0);
+                                    }
 
-                            events.chunk.get(events.chunk.size() - 1).mToken = events.end;
+                                    store.storeRoomEvents(roomId, events, EventTimeline.Direction.BACKWARDS);
+                                }
 
-                            Event firstReturnedEvent = events.chunk.get(0);
-                            if ((oldestEvent != null) && (firstReturnedEvent != null)
-                                    && TextUtils.equals(oldestEvent.eventId, firstReturnedEvent.eventId)) {
-                                events.chunk.remove(0);
+                                Log.d(LOG_TAG, "## backPaginate() succeed : roomId " + roomId + " token " + token + " got " + events.chunk.size());
+                                callback.onSuccess(events);
                             }
+                        }
 
-                            store.storeRoomEvents(roomId, events, EventTimeline.Direction.BACKWARDS);
+                        private void logErrorMessage(String expectedToken, String errorMessage) {
+                            Log.e(LOG_TAG, "## backPaginate() failed : roomId " + roomId + " token " + token + " expected " + expectedToken + " with " + errorMessage);
                         }
 
-                        Log.d(LOG_TAG, "## backPaginate() succeed : roomId " + roomId + " token " + token + " got " + events.chunk.size());
-                        callback.onSuccess(events);
-                    }
-                }
+                        @Override
+                        public void onNetworkError(Exception e) {
```

Comment:
GitHub comparison tools is so limited... It's seems that there is only new spaces and this tool consider the whole code has changed. Or is there something I miss?

Reply:
Didn't touch this code, automatic reformat I guess